src/fresh_merge.rs

```diff
@@ -3,531 +3,535 @@ use log::info;
-  /// Returns true if atomic gate is ushel the limit
-  fn check_limit(gate: &AtomicUsize, limit: usize) -> bool {
-    gate.load(Ordering::Acquire) < limit
-  }
+    let hpg = Arc::clone(&holding_pen_gate);
+    let coorindator_handle = thread::spawn(move || {
+        let mut pos = 0usize;
+        loop {
+            // Queue items for work until the buffer limit is reached
+            if check_limit(&hpg, merge_buffer_limit) {
+                match next_hash_of_item_to_merge(&mut index_holder) {
+                    Some(items_to_merge) => {
+                        offset_tx
+                            .send((pos, items_to_merge))
+                            .expect("Should be able to send the message");
+                        pos += 1;
+                    }
+                    // No more work left to do!
+                    None => break,
+                }
+            }
+            // Park this coordinator when the limit is reached
+            else {
+                while !check_limit(&hpg, merge_buffer_limit) {
+                    thread::sleep(Duration::from_millis(20));
+                }
+            }
+        }
+    });
 
-  /// Trigger parked coordinator threads to continue running
-  fn unpark_coords(coords: &Vec<Thread>) {
-    for t in coords {
-      Thread::unpark(&t);
+    threads.push(coorindator_handle);
+
+    //// Worker thread section
+    ////
+    let (merged_tx, merged_rx) = flume::bounded(200_000);
+    for thread_num in 0..20 {
+        let rx = offset_rx.clone();
+        let tx = merged_tx.clone();
+
+        let processor_handle = thread::spawn(move || {
+            let mut cnt = 0usize;
+            while let Ok((position, items_to_merge)) = rx.recv() {
+                let mut to_merge = vec![];
+                let mut purls = vec![];
+                for (offset, cluster) in &items_to_merge {
+                    let merge_final = cluster
+                        .item_from_item_offset(offset)
+                        .expect("Should get the item");
+                    merge_final
+                        .connections
+                        .iter()
+                        .filter(|v| v.1.starts_with("pkg:"))
+                        .for_each(|v| purls.push(v.1.to_string()));
+
+                    to_merge.push(merge_final);
+                }
+
+                let mut top = to_merge.pop().unwrap();
+
+                for i in to_merge {
+                    top = top.merge(i);
+                }
+
+                let cbor_bytes =
+                    serde_cbor::to_vec(&top).expect("Should be able to encode an item");
+
+                tx.send(ItemOrPurl::Item {
+                    pos: position,
+                    item: top,
+                    cbor_bytes,
+                    merged: items_to_merge.len() - 1,
+                    purls,
+                })
+                .expect("Should send message");
+                cnt += 1;
+                if false && cnt % 500_000 == 0 {
```

Comment:
why `if false` instead of either commenting out or just deleting it?

Reply:
Dunno... legacy